Index: common/cmd_bootm.c
===================================================================
RCS file: /cvsroot/u-boot/u-boot/common/cmd_bootm.c,v
retrieving revision 1.48
diff -C3 -r1.48 cmd_bootm.c
*** common/cmd_bootm.c	15 Oct 2005 17:08:09 -0000	1.48
--- common/cmd_bootm.c	8 Mar 2006 10:05:45 -0000
***************
*** 116,121 ****
--- 116,123 ----
  extern int do_bdinfo ( cmd_tbl_t *cmdtp, int flag, int argc, char *argv[]);
  #endif
  
+ static ulong g_image_length = 0;
+ 
  #ifdef CONFIG_PPC
  static boot_os_Fcn do_bootm_linux;
  #else
***************
*** 150,161 ****
  	ulong	addr;
  	ulong	data, len, checksum;
  	ulong  *len_ptr;
! 	uint	unc_len = 0x400000;
  	int	i, verify;
  	char	*name, *s;
  	int	(*appl)(int, char *[]);
  	image_header_t *hdr = &header;
  
  	s = getenv ("verify");
  	verify = (s && (*s == 'n')) ? 0 : 1;
  
--- 152,168 ----
  	ulong	addr;
  	ulong	data, len, checksum;
  	ulong  *len_ptr;
! 	uint	unc_len = 0x00A00000;
  	int	i, verify;
  	char	*name, *s;
  	int	(*appl)(int, char *[]);
  	image_header_t *hdr = &header;
  
+ 	s = getenv("zip_window_size");
+ 
+ 	if (s)
+ 	    unc_len = simple_strtoul(s, NULL, 16);
+ 	
  	s = getenv ("verify");
  	verify = (s && (*s == 'n')) ? 0 : 1;
  
***************
*** 369,374 ****
--- 376,383 ----
  	puts ("OK\n");
  	SHOW_BOOT_PROGRESS (7);
  
+ 	g_image_length = len;
+ 
  	switch (hdr->ih_type) {
  	case IH_TYPE_STANDALONE:
  		if (iflag)
***************
*** 523,528 ****
--- 532,538 ----
  #ifdef CONFIG_OF_FLAT_TREE
  	char	*of_flat_tree;
  #endif
+ 	ulong	boot_stack_size = 0x40000; /* 256k */
  
  	if ((s = getenv ("initrd_high")) != NULL) {
  		/* a value of "no" or a similar string will act like 0,
***************
*** 557,565 ****
  
  	debug ("## Current stack ends at 0x%08lX ", sp);
  
! 	sp -= 2048;		/* just to be sure */
! 	if (sp > CFG_BOOTMAPSZ)
! 		sp = CFG_BOOTMAPSZ;
  	sp &= ~0xF;
  
  	debug ("=> set upper limit to 0x%08lX\n", sp);
--- 567,576 ----
  
  	debug ("## Current stack ends at 0x%08lX ", sp);
  
! 	if ((s = getenv("boot_stack_size")) != NULL)
! 	    boot_stack_size = simple_strtoul(s, NULL, 16); 
! 
! 	sp = g_image_length + boot_stack_size + 0xF;
  	sp &= ~0xF;
  
  	debug ("=> set upper limit to 0x%08lX\n", sp);
Index: include/mpc8260.h
===================================================================
RCS file: /cvsroot/u-boot/u-boot/include/mpc8260.h,v
retrieving revision 1.5
diff -C3 -r1.5 mpc8260.h
*** include/mpc8260.h	15 Apr 2004 18:22:43 -0000	1.5
--- include/mpc8260.h	8 Mar 2006 10:05:46 -0000
***************
*** 272,277 ****
--- 272,278 ----
  #define HRCW_CS10PC01	0x00000040	/* - " -			*/
  #define HRCW_CS10PC10	0x00000080	/* - " -			*/
  #define HRCW_CS10PC11	0x000000c0	/* - " -			*/
+ #define HCRW_PCI_MODCK  0x00000010
  #define HRCW_MODCK_H0000 0x00000000	/* High-order bits of MODCK Bus	*/
  #define HRCW_MODCK_H0001 0x00000001	/* - " -			*/
  #define HRCW_MODCK_H0010 0x00000002	/* - " -			*/
Index: include/configs/MPC8260ADS.h
===================================================================
RCS file: /cvsroot/u-boot/u-boot/include/configs/MPC8260ADS.h,v
retrieving revision 1.29
diff -C3 -r1.29 MPC8260ADS.h
*** include/configs/MPC8260ADS.h	25 Sep 2005 23:07:37 -0000	1.29
--- include/configs/MPC8260ADS.h	8 Mar 2006 10:05:46 -0000
***************
*** 246,254 ****
  /* this must be included AFTER the definition of CONFIG_COMMANDS (if any) */
  #include <cmd_confdefs.h>
  
! #define CONFIG_BOOTDELAY	5		/* autoboot after 5 seconds */
! #define CONFIG_BOOTCOMMAND	"bootm fff80000"	/* autoboot command */
! #define CONFIG_BOOTARGS		"root=/dev/mtdblock2"
  
  #if (CONFIG_COMMANDS & CFG_CMD_KGDB)
  #undef	CONFIG_KGDB_ON_SMC		/* define if kgdb on SMC */
--- 246,254 ----
  /* this must be included AFTER the definition of CONFIG_COMMANDS (if any) */
  #include <cmd_confdefs.h>
  
! #define CONFIG_BOOTDELAY	5			/* autoboot after 5 seconds */
! #define CONFIG_BOOTCOMMAND	"bootm ff890000"	/* autoboot command */
! #define CONFIG_BOOTARGS		"console=ttyS0,115200 root=/dev/ram0 rw"
  
  #if (CONFIG_COMMANDS & CFG_CMD_KGDB)
  #undef	CONFIG_KGDB_ON_SMC		/* define if kgdb on SMC */
***************
*** 333,344 ****
--- 333,353 ----
  #define CFG_INIT_SP_OFFSET	CFG_GBL_DATA_OFFSET
  
  #ifdef CFG_LOWBOOT
+ #if CONFIG_ADSTYPE == CFG_8272ADS
+ /* MPC8272ADS Flash HRCW 0x0E74B20A */
+ #define CFG_HRCW_MASTER (   ( HRCW_BPS11 | HRCW_CIP )                      |\
+ 				( HRCW_L2CPC01 | HRCW_DPPC11 | HRCW_ISB100 )    |\
+ 				( HRCW_BMS | HRCW_MMR11 | HRCW_LBPC00 | HRCW_APPC10 ) |\
+ 				( HCRW_PCI_MODCK | HRCW_MODCK_H1010 )             \
+ 		)
+ #else
  /* PQ2FADS flash HRCW = 0x0EB4B645 */
  #define CFG_HRCW_MASTER (   ( HRCW_BPS11 | HRCW_CIP )			    |\
  			    ( HRCW_L2CPC10 | HRCW_DPPC11 | HRCW_ISB100 )    |\
  			    ( HRCW_BMS | HRCW_MMR11 | HRCW_LBPC01 | HRCW_APPC10 ) |\
  			    ( HRCW_CS10PC01 | HRCW_MODCK_H0101 )	     \
  			)
+ #endif
  #else
  /* PQ2FADS BCSR HRCW = 0x0CB23645 */
  #define CFG_HRCW_MASTER (   ( HRCW_BPS11 | HRCW_CIP )			    |\
***************
*** 346,352 ****
  			    ( HRCW_BMS | HRCW_APPC10 )			    |\
  			    ( HRCW_MODCK_H0101 )			     \
  			)
! #endif
  /* no slaves */
  #define CFG_HRCW_SLAVE1 0
  #define CFG_HRCW_SLAVE2 0
--- 355,361 ----
  			    ( HRCW_BMS | HRCW_APPC10 )			    |\
  			    ( HRCW_MODCK_H0101 )			     \
  			)
! #endif /*CFG_LOWBOOT*/
  /* no slaves */
  #define CFG_HRCW_SLAVE1 0
  #define CFG_HRCW_SLAVE2 0
